sites/friday/src/get-started/here/make-changes.md

```diff
@@ -0,0 +1,118 @@
+   This enables you, as a developer, to use a normal Git workflow (`git add . && git commit -m "message" && git push`) to deploy your environment—with no need to connect to the {{% vendor/name %}} Console.
+
+   Assuming you've already activated the source integration feature on your project, create a new branch:
+   ```bash {location="Terminal"}
+   git checkout -b feat-a && git push -u origin feat-a
+   ```
+   This creates a new local `feat-a` Git branch based on the `main` Git branch
+   and a related environment on {{< vendor/name >}}.
+   The new environment is inactive by default. When you would activate it, it will inherit the data (service data and assets) of its parent environment (the Production environment here).
+
+   {{< note >}}
+   When your local feature is ready, [activate your environment](/environments/deactivate-environment.md#reactivate-an-environment).
+   {{< /note >}}
+
+{{< /codetabs >}}
+
+{{< note title="Warnings" theme="warning">}}
+   **First**, you are only charged for your [activated environments](/environments/deactivate-environment.md#reactivate-an-environment).
+   So, to create an inactive environment waiting for you to be ready to test, run the following command:
+   ```bash {location="Terminal"}
+   git branch checkout -b feat-a && git push -u upsun feat-a
+   ```
+   **Secondly**, if you have set up a [source integration](/integrations/source/_index.md) and have enabled `prune_branches` (`true` by default),
+   the `upsun branch` command fails at the end of the deployment.
+   This is due to the integration syncing with the source and deleting the new branch since it doesn't exist in the canonical source:
+   ```bash {location="Terminal"}
+   fatal: couldn't find remote ref feat-a
+   ```
+{{< /note >}}
+
+## 2. Make changes to your project.
+
+Depending on the stack you're using, change something within your source code.
+
+## 3. Commit your changes
+
+```bash {location="Terminal"}
+git add index.js
+git commit -m "Update source code"
+```
+
+## 4. Deploy your changes
+
+Deploy your changes to the `feat-a` environment:
+
+```bash {location="Terminal"}
+{{% vendor/cli %}} push
+```
+
+Note that each environment has its own domain name.
+To open the URL of your new environment, run the following command:
+
+```bash {location="Terminal"}
+{{% vendor/cli %}} environment:url --primary
+```
+{{< note title="Warnings" theme="warning">}}
+If your environment is inactive, you need to [activate it](/environments/deactivate-environment.md#reactivate-an-environment) first, using the following:
+```bash {location="Terminal"}
+{{% vendor/cli %}} environment:activate
+```
+{{< /note >}}
+
+## 5. Iterate
+Iterate by changing the code, committing, and deploying.
+When satisfied with your changes, merge them to the main branch,
+and remove the feature branch:
+
+```bash {location="Terminal"}
+{{% vendor/cli %}} merge
```

Comment:
we're going to need to split this into codetabs again for non-integrated vs integrated since the flow can be different, and will break if the default/production branch is protected in the source control management system (ie GitHub)

Reply:
i'm confused why this was marked resolved since it is still relevant. We're still splitting the path currently between `Upsun Git repository` and `Using third party provider` in step 1, so there's still a significant probability that attempting to perform a merge via our cli tool at step 5 will fail when using a 3rd party integration.

Reply:
ok, i try to split any git/upsun snippet that could lead to errors/confusion using codetabs.